cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/co/cask/cdap/templates/etl/api/batch/BatchSource.java

```diff
@@ -41,4 +46,28 @@ public void configure(StageConfigurer configurer) {
    * @param context {@link BatchSourceContext}
    */
   public abstract void prepareJob(BatchSourceContext context);
+
+  protected long parseFrequency(String frequencyStr) {
```

Comment:
this shouldn't be in BatchSource as it's not related to source. Can pull it out to some common helper class.